src/color.rs

```diff
@@ -1,7 +1,7 @@
 /// A terminal color.
 pub trait Color {
-    /// Convert this to its ANSI value.
-    fn to_ansi_val(self) -> u8;
+    /// Get an escape code corresponding to the color value.
+    fn to_escape_code(self, is_background: bool) -> String;
```

Comment:
I am very opposed to this change. Having allocations for something as simple as changing colors is something termion tries to avoid. I think the colors could be amended to be structs implementing the `Display` trait, which is something I'm working on.